### ISSUE TITLE
Onion skin function

### ISSUE THREAD

**Issue by GeminiCrafterMan** [kirjavascript/Flex2#50]
I know I've suggested this on Discord like three times, but it really would make aligning sprites slightly easier. Might try to do it myself and submit a pull request, but I have zero experience working with Electron, so that's not a guarantee.

[Here's the example from the Wikipedia article for onion skinning, just to make sure you know what I'm talking about this time.](https://upload.wikimedia.org/wikipedia/commons/5/5e/Onion_skin.png)

Maybe right-clicking on a frame or pressing 'o' in the editor could do it? Add a box for what frame to display in the control list, maybe?

**Comment by kirjavascript** [comment]
ah I'm not really active on discord so wouldn't've seen the messages.

this is a good solution to the lining up problem (MainMemory suggested having a masking image in addition, but this is easier) and it'd be nice to use it. the UI / controls just need to be considered.

if you don't end up implementing it I will next time I'm working on this project

**Comment by kirjavascript** [comment]
maybe this should be an option on import

**Comment by iojnekns** [comment]
I associate onion skinning with drawing of sprites.  For lining them up, a configurable grid may be a more direct solution?

**Comment by kirjavascript** [comment]
yeah, there a different ways to approach the UX for this can I can't settle on what is best

you can have an additional 'mask' file to define where the centres are but I don't like that either

what do you mean by a configurable grid? something you draw on the image you're importing or something else?

**Comment by iojnekns** [comment]
you currently only provide one line per axis ('guidelines') rather than a grid overlay.  a full, configurable grid has more points of reference, easier to line things up
onion skin is the wrong fit for the stated issue imo, the frame blending makes it ambiguous which pixel belongs to the current frame vs the blended one.

**Comment by kirjavascript** [comment]
ah I see!

> onion skin is the wrong fit for the stated issue imo, the frame blending makes it ambiguous which pixel belongs to the current frame vs the blended one.

you could work around this by giving the previous frame a blue tint or border or outline something

**Comment by kirjavascript** [comment]
really, I want explore a solution for doing this at the point of import though, rather than add something to the editor itself

**Comment by iojnekns** [comment]
if the user is preparing the graphics for alignment at the point of import anyway, my suggestion would be to set a flag during the import process that lets the user specify X/Y frames for their spritesheet. they can align each frame to the center of its 'cell' in their graphics program (probably using their graphics program's configurable grid) and they will be imported predictably.

configurable grid would be a nice to have regardless.

**Comment by kirjavascript** [comment]
yeah the grid would be good regardless

I'm not sure just centering each sprite is enough: they need to be aligned relative to each other, not some centre. this often means the floor, or some specific angle for rotated sprites

you could still use the same concept you mentioned and specify some origin instead of centering, maybe

ideally I'd like to get some input from people that do this a lot on what they think, but I had no takers so far when asking

**Comment by iojnekns** [comment]
the user can put the sprites anywhere they like within the respective cell - they don't have to center it.

fyi, this is how eg aseprite / libresprite exports animations to spritesheet and it's how eg godot expects to import animation frames from spritesheet.
https://www.aseprite.org/docs/sprite-sheet/
https://docs.godotengine.org/en/stable/tutorials/2d/2d_sprite_animation.html#sprite-sheet-with-animatedsprite2d
